File: web/lib/services/admin.service.ts
import { apiClient, API_CONFIG } from "@/lib/api-client";
import type {
  AdminDashboardStats,
  AdminUser,
  AdminProduct,
  AdminOrder,
  AdminCategory,
  UpdateUserStatusRequest,
} from "@/lib/types/admin";

export class AdminService {
  // ── Dashboard ────────────────────────────────────────────────
  async getDashboardStats(): Promise<AdminDashboardStats> {
    return apiClient.get<AdminDashboardStats>(
      API_CONFIG.endpoints.admin.dashboard,
    );
  }

  // ── Users ────────────────────────────────────────────────────
  async getAllUsers(): Promise<AdminUser[]> {
    return apiClient.get<AdminUser[]>(API_CONFIG.endpoints.admin.users);
  }

  async setUserStatus(userId: string, active: boolean): Promise<AdminUser> {
    const body: UpdateUserStatusRequest = { active };
    return apiClient.put<AdminUser>(
      `${API_CONFIG.endpoints.admin.users}/${userId}/status`,
      body,
    );
  }

  async deleteUser(userId: string): Promise<void> {
    return apiClient.delete(`${API_CONFIG.endpoints.admin.users}/${userId}`);
  }

  // ── Products ─────────────────────────────────────────────────
  async getAllProducts(): Promise<AdminProduct[]> {
    return apiClient.get<AdminProduct[]>(API_CONFIG.endpoints.admin.products);
  }

  async deleteProduct(productId: string): Promise<void> {
    return apiClient.delete(
      `${API_CONFIG.endpoints.admin.products}/${productId}`,
    );
  }

  // ── Orders ──────────────────────────────────────────────────
  async getAllOrders(): Promise<AdminOrder[]> {
    return apiClient.get<AdminOrder[]>(API_CONFIG.endpoints.admin.orders);
  }

  async updateOrderStatus(
    orderId: string,
    status: string,
  ): Promise<AdminOrder> {
    return apiClient.patch<AdminOrder>(
      `${API_CONFIG.endpoints.admin.orders}/${orderId}/status?status=${status}`,
      {},
    );
  }

  // ── Categories ───────────────────────────────────────────────
  async getAllCategories(): Promise<AdminCategory[]> {
    return apiClient.get<AdminCategory[]>(
      API_CONFIG.endpoints.admin.categories,
    );
  }

  async createCategory(
    data: Omit<AdminCategory, "id" | "createdAt">,
  ): Promise<AdminCategory> {
    return apiClient.post<AdminCategory>(
      API_CONFIG.endpoints.admin.categories,
      data,
    );
  }

  async updateCategory(
    id: string,
    data: Omit<AdminCategory, "id" | "createdAt">,
  ): Promise<AdminCategory> {
    return apiClient.put<AdminCategory>(
      `${API_CONFIG.endpoints.admin.categories}/${id}`,
      data,
    );
  }

  async deleteCategory(id: string): Promise<void> {
    return apiClient.delete(`${API_CONFIG.endpoints.admin.categories}/${id}`);
  }
}

export const adminService = new AdminService();

File: ecom.api/src/main/java/aditi/wing/ecom/api/domain/admin/service/AdminService.java
package aditi.wing.ecom.api.domain.admin.service;

import java.util.List;
import java.util.UUID;

import aditi.wing.ecom.api.domain.admin.dto.AdminCategoryDto;
import aditi.wing.ecom.api.domain.admin.dto.AdminDashboardStatsDto;
import aditi.wing.ecom.api.domain.admin.dto.AdminUserDto;
import aditi.wing.ecom.api.domain.orders.dto.OrderResponse;
import aditi.wing.ecom.api.domain.orders.enums.OrderStatus;
import aditi.wing.ecom.api.domain.seller.dto.ProductResponseDto;

public interface AdminService {

    /** Platform-wide dashboard statistics */
    AdminDashboardStatsDto getDashboardStats();

    /** User management */
    List<AdminUserDto> getAllUsers();

    AdminUserDto setUserStatus(UUID userId, boolean active);

    void deleteUser(UUID userId);

    /** Product management */
    List<ProductResponseDto> getAllProducts();

    void deleteProduct(UUID productId);

    /** Order management */
    List<OrderResponse> getAllOrders();

    OrderResponse updateOrderStatus(UUID orderId, OrderStatus status);

    /** Category management */
    List<AdminCategoryDto> getAllCategories();

    AdminCategoryDto createCategory(AdminCategoryDto dto);

    AdminCategoryDto updateCategory(UUID id, AdminCategoryDto dto);

    void deleteCategory(UUID id);
}
